src/config.rs

```diff
@@ -0,0 +1,182 @@
+use std::env::VarError;
+use std::error::Error;
+use std::fmt;
+use std::str::FromStr;
+
+const ENVIRONMENT_VARIABLE_PREFIX: &str = "MONITORBOT_";
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    // http server port to bind to
+    pub port: u16,
+    // github api tokens to collect rate limit statistics
+    pub gh_rate_limit_tokens: String,
+    // github rate limit stats data cache refresh rate frequency (in seconds)
+    pub gh_rate_limit_stats_cache_refresh: u64,
+}
+
+impl Config {
+    pub fn from_env() -> Result<Self, ConfigError> {
+        Ok(Self {
+            port: Self::default_env("PORT", 3001)?,
+            gh_rate_limit_tokens: Self::require_env("RATE_LIMIT_TOKENS")?,
+            gh_rate_limit_stats_cache_refresh: Self::default_env::<u64>(
+                "GH_RATE_LIMIT_STATS_REFRESH",
+                120u64,
+            )?,
+        })
+    }
+
+    fn maybe_env<T: FromStr>(name: &str) -> Result<Option<T>, ConfigError> {
+        match std::env::var(format!("{}{}", ENVIRONMENT_VARIABLE_PREFIX, name)) {
+            Ok(val) => match val.parse() {
+                Ok(v) => Ok(Some(v)),
+                _ => Err(ConfigError(format!(
+                    "the {} environment variable has invalid content",
+                    name
+                ))),
```

Comment:
You are erasing the actual error message here. If you don't include the error message itself, debugging why the variable had invalid content could be tricky. In the promote-release implementation this used anyhow's `.context()`, which adds another "layer" of error on top of the one returned by the `parse` method.

Personally I would just use anyhow here. It adds nice utilities like `context`, and requires less boilerplate.

Reply:
sure, I'll bring anyhow crate then. 👍